analysis/real.lean

```diff
@@ -350,4 +350,58 @@ compact_of_totally_bounded_is_closed
   (real.totally_bounded_Icc a b)
   (is_closed_inter (is_closed_ge' a) (is_closed_le' b))
 
+open real
+
+lemma real.intermediate_value {f : ℝ → ℝ} {a b t : ℝ}
+  (hf : ∀ x, a ≤ x → x ≤ b → tendsto f (nhds x) (nhds (f x)))
```

Comment:
You are assuming full continuity at `a` and `b` instead of one-sided continuity (i.e., continuity within `[a,b]`), that would be the natural condition here.

Reply:
what @sgouezel  wants is:
`hf : ∀ x, a ≤ x → x ≤ b → tendsto f (inf (nhds x) (principal (Icc a b))) (nhds (f x))`

Reply:
@johoelzl What do you think of defining a filter `nhds_within S x = inf (nhds x) (principal S)` and use it to define continuity within S (not in this pull request, of course!). Or is it better design to always go first to the subtype `S` and then use `nhds x` for the induced topology? (Of course, the two are equivalent). With my Isabelle-biased point of view, I tend to prefer the first approach, avoiding subtypes when possible.

Reply:
My hope is that the `subtype` approach is working. So also here it would be an option to write:

`hf : ∀ x:Icc a b, tendsto (λx:Icc a b, f x.1) (nhds x) (nhds (f x.1))`

Reply:
I think if you really want to pursue this route you should just treat `Icc a b` as a topological space in its own right, and `f` is a continuous function on that domain. We will probably need a lot of lemmas about the unit interval as a topological space anyway, since it plays so prominently in path connectedness and the fundamental group.